src/app/file/file-list/file-list.component.ts

```diff
@@ -209,27 +209,14 @@ export class FileListComponent implements OnInit, OnDestroy {
         }));
     }
     private removeFile(fileName: string): void {
-        this.confirm(`Do you want to delete "${fileName}"?`, 'Delete a file', 'Delete')
-            .pipe( switchMap(it => this.fileService.removeFile(this.path.absolutePath(fileName))))
+        this.modalService.confirm(`Do you want to delete "${fileName}"?`, 'Delete a file', 'Delete')
```

Comment:
you can do confirm().filter() .. instead of checking value in the switchMap

Reply:
if you return Observable.of({})  data will be reloaded by this.loadData() call, but you do not want it